pkg/storegateway/bucket.go

```diff
@@ -511,44 +512,24 @@ func (s *BucketStore) removeBlock(id ulid.ULID) (returnErr error) {
 }
 
 func (s *BucketStore) removeAllBlocks() error {
-	// Build a list of blocks to remove.
-	s.blocksMx.Lock()
-	blockIDs := make([]ulid.ULID, 0, len(s.blocks))
-	for id := range s.blocks {
-		blockIDs = append(blockIDs, id)
-	}
-	s.blocksMx.Unlock()
-
-	// Close all blocks.
 	errs := multierror.New()
-
-	for _, id := range blockIDs {
+	s.blocks.Range(func(key, _ any) bool {
```

Comment:
This seems like it's changing behavior a bit. Previously, we'd lock `s.blocks` to build a list of IDs before making changes to `s.blocks`. Now, we're modifying `s.blocks` as we're iterating it. It seems like this isn't forbidden by `sync.Map` but I'm not sure it's correct.